raft.c

```diff
@@ -1339,6 +1411,91 @@ void RaftReqHandleQueue(uv_async_t *handle)
  * Implementation of specific request types.
  */
 
+static void handleTransferLeaderComplete(raft_server_t *raft, raft_transfer_state_e state)
+{
+    raft_node_id_t transfer_id = raft_get_transfer_leader(raft);
+    raft_node_id_t leader = raft_get_leader_id(raft);
+
+    if (!redis_raft.transfer_req) {
+        LOG_ERROR("leader transfer update: but no req to correlate it to!");
+        return;
+    }
+
+    if (!transfer_id) {
+        LOG_ERROR("leader transfer: libraft didn't return a node id we were transferring to");
+        RedisModule_ReplyWithError(redis_raft.transfer_req->ctx, "ERR: no record of a leader transfer");
+        goto exit;
+    }
+
+    switch (state) {
+        case RAFT_STATE_LEADERSHIP_TRANSFER_EXPECTED_LEADER:
+            RedisModule_ReplyWithSimpleString(redis_raft.transfer_req->ctx, "OK");
+            break;
+        case RAFT_STATE_LEADERSHIP_TRANSFER_UNEXPECTED_LEADER:
+            RedisModule_ReplyWithError(redis_raft.transfer_req->ctx, "ERR: transferred to unexpected leader");
```

Comment:
This sounds a bit like we transferred to a wrong node by mistake. Is it better to say something like "ERR another node elected as leader"